### DISCUSSION TITLE
Manually adding re-invested dividends

### DISCUSSION THREAD

**Discussion by jswantekdds** [investbrainapp/investbrain#35] [Q&A] [answered]
I'm loving the project!  I'm just wondering how dividends work. I see there's an area where I can mark a holding as "reinvest dividends" and there's also a space to see dividends for a holding. Can I mark a transaction as a reinvested dividend? I see that there's a space for that on the import spreadsheet but I'm not seeing anything in the GUI. Thanks!

**Comment by hackeresq** [comment, marked as answer]
Hi @jswantekdds, moving this to a discussion since this isn't really an issue. 

But this is a fantastic question! Currently in the GUI there is no option to mark transactions as a "reinvestment". Adding additional options to the transaction dialog feels like a slippery slope. So there are no plans to add that additional toggle to the transactions dialog.

However, you can still mark existing transactions as re-investments! Here's what I would recommend:  "export" your transactions, then manually mark each re-invested dividend transaction in Excel (seems like you've already seen that column).  Once you've done that, you can re-import the transactions. Since imports are performed using an "upsert", this will update the existing transactions to reflect them as re-invested dividends.

Hope that makes sense?

**Reply by jswantekdds** [reply]
Looks like that worked!  Thanks!  FYI and for clarity, I had to mark the reinvested dividend transaction as "true" and the other transactions as "false".  If I left the 'false' ones blank then it generated a ton of weird reinvested transactions.

**Reply by hackeresq** [reply]
Thanks for confirming that worked!  Interesting, I'll take a look at that "blank" cell behavior.

**Comment by EnduringGuerila** [comment]
I just set up Investbrain and don't have any existing reinvestments to reference. 
I added a few transactions and did an export to get a simple template. 

How should the reinvestment rows look?
What is the minimum amount of columns needed for importing?
How do I add stock splits?

I'm planning on importing all the transactions via csv and I think it'd be great if someone made a guide to help new users import in bulk. 
Vanguard isn't letting me export a csv for transactions older than 18 months but I can copy/paste my 5 years of transactions into a spreadsheet to clean up.

**Reply by EnduringGuerila** [reply]
[2025_11_08_investbrain_backup.xlsx](https://github.com/user-attachments/files/23434925/2025_11_08_investbrain_backup.xlsx)
my current sheet (wip)

lines 10-300 are manually added on the transactions sheet. 

For reinvestments, do I just mark them as 'BUY' and then 'TRUE' in the Reinvested column and all the blank fields as 'FALSE' to avoid an apparent issue?

I had started logging dividend reinvestments in Marketwatch as $0.00001 so I could log them as gains instead of normal buys, and I had duplicated that here for rows 4 and 9. I am guessing once properly marked as reinvestments, I can put in the actual price?

No idea what to do with this either: 
SFGYY Sony Corporate Spin Off 35 shares. (link 299 I think)
List is as a buy at $0?
[2025_11_08_investbrain_backup.xlsx](https://github.com/user-attachments/files/23434925/2025_11_08_investbrain_backup.xlsx)

**Reply by EnduringGuerila** [reply]
I did a test import for one reinvest transaction and it really didn't like the date formatted fields, excel wouldn't quit fucking with them so I had to mark it as a text field to get the same formatting. 

My 3 small tests went well so...

I then imported this file just to wing it, which went through but caused some errors somewhere and now I cannot view my portfolio or transactions pages.
[2025_11_08_investbrain_import_nosplit.xlsx](https://github.com/user-attachments/files/23435084/2025_11_08_investbrain_import_nosplit.xlsx)

```
2025-11-08 23:40:36 App\Jobs\BackupImportJob ....................... 1s DONE
  2025-11-08 23:40:36 App\Console\Commands\RefreshMarketData ......... RUNNING
  2025-11-08 23:40:37 App\Notifications\ImportSucceededNotification .. RUNNING
  2025-11-08 23:40:37 App\Notifications\ImportSucceededNotification  20.48ms DONE
  2025-11-08 23:40:38 App\Console\Commands\RefreshMarketData ......... 2s DONE
  2025-11-08 23:40:38 Closure (fn () => \Illuminate\Support\Facades\Artisan::call(\App\Console\Commands\RefreshDividendData::class, ['--user' => $this->backupImportModel->user_id, '--force' => true]):2)  RUNNING
  2025-11-08 23:40:38 Closure (fn () => \Illuminate\Support\Facades\Artisan::call(\App\Console\Commands\RefreshDividendData::class, ['--user' => $this->backupImportModel->user_id, '--force' => true]):2)  189.35ms FAIL
   INFO  No scheduled commands are ready to run.  

   INFO  No scheduled commands are ready to run.  

   INFO  No scheduled commands are ready to run.  

   INFO  No scheduled commands are ready to run.  

   INFO  No scheduled commands are ready to run.  

   INFO  No scheduled commands are ready to run.  

   INFO  No scheduled commands are ready to run.  

   INFO  No scheduled commands are ready to run.  

   INFO  No scheduled commands are ready to run.  

   INFO  No scheduled commands are ready to run.  

   INFO  No scheduled commands are ready to run.  

   INFO  No scheduled commands are ready to run.  

   INFO  No scheduled commands are ready to run.  

   INFO  No scheduled commands are ready to run.  

   INFO  No scheduled commands are ready to run.  

   INFO  No scheduled commands are ready to run.  

   INFO  No scheduled commands are ready to run.  

   INFO  No scheduled commands are ready to run.  

   INFO  No scheduled commands are ready to run.  

  2025-11-09 00:00:00 Running ['artisan' refresh:split-data] ......... 2s DONE
  ⇂ Refresh split data from data provider  
  2025-11-09 00:00:03 Running ['artisan' refresh:currency-data]  969.41ms DONE
  ⇂ Refresh currency data from data provider
```
